<?xml version="1.0"?>
<launch>

    <!-- Scout files -->

    <!-- initial pose -->
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <arg name="scout_name" default="robot"/>
    <arg name="robot_namespace" default=""/>

    <arg name="pi" default="0.00"/>

    <arg name="robot_joint_states" default=
    " -J $(arg scout_name)::shoulder_pan_joint $(arg pi) 
      -J $(arg scout_name)::shoulder_lift_joint $(arg pi)
      -J $(arg scout_name)::elbow_joint $(arg pi)
      -J $(arg scout_name)::wrist_1_joint $(arg pi)
      -J $(arg scout_name)::wrist_2_joint $(arg pi)
      -J $(arg scout_name)::wrist_3_joint $(arg pi)"/>

    <!-- these are the arguments you can pass this launch file, for example paused:=true -->
    <arg name="paused" default="true" />
    <arg name="use_sim_time" default="true" />
    <arg name="gui" default="true" />
    <arg name="headless" default="false" />
    <arg name="debug" default="true" />

    <arg name="urdf_extras" default="$(find scout_description)/urdf/empty.urdf"/>

    <include file="$(find scout_description)/launch/scout_v2_ur3e_stock.launch">
        <arg name="robot_namespace" value="$(arg robot_namespace)" />
    </include>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find scout_gazebo_sim)/config/scout_v2_control.yaml" command="load"/>
            <!-- <rosparam command="load" file="$(find scout_control)/config/scout2_control.yaml"/>
                -->
    
    <!-- load the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" 
    args="scout_state_controller scout_motor_fr_controller scout_motor_fl_controller scout_motor_rl_controller scout_motor_rr_controller"/>

    <node name="scout_skid_steer_controller" pkg="scout_gazebo_sim" type="scout_skid_steer_controller">
        <param name="robot_namespace" type="string" value="$(arg robot_namespace)" />
    </node>

    <!-- Spawn model with initial args -->
    <node name="spawn_scout_model" pkg="gazebo_ros" type="spawn_model" args="-x $(arg x)
      -y $(arg y)
      -z $(arg z)
      -Y $(arg yaw)
      -unpause
      -urdf
      -param robot_description
      -model $(arg scout_name)$(arg robot_namespace)
      $(arg robot_joint_states)" 
      output="screen"
      respawn="false"
    /> 

</launch>